File: Source/Padutronics.Windows.Win32.Api/Dxgi/IDXGISwapChain.cs
using Padutronics.Windows.Win32.Api.DxgiFormat;
using System;
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.Dxgi;

[Guid("310d36a0-d2e7-4c0a-aa04-6a9d23b8886a")]
public interface IDXGISwapChain : IDXGIDeviceSubObject
{
    void GetBuffer(int Buffer, Guid riid, out nint ppSurface);
    void Present(int SyncInterval, DXGI_PRESENT Flags);
    void ResizeBuffers(int BufferCount, int Width, int Height, DXGI_FORMAT NewFormat, DXGI_SWAP_CHAIN_FLAG SwapChainFlags);
}

File: Source/Padutronics.Windows.Win32.Api.CWrapper/Dxgi/IDXGISwapChain.cpp
#include "IDXGISwapChain.hpp"

auto IDXGISwapChain_GetBuffer(IDXGISwapChain& _this, UINT Buffer, REFIID riid, void** ppSurface) -> HRESULT
{
    return _this.GetBuffer(Buffer, riid, ppSurface);
}

auto IDXGISwapChain_Present(IDXGISwapChain& _this, UINT SyncInterval, UINT Flags) -> HRESULT
{
    return _this.Present(SyncInterval, Flags);
}

auto IDXGISwapChain_ResizeBuffers(IDXGISwapChain& _this, UINT BufferCount, UINT Width, UINT Height, DXGI_FORMAT NewFormat, UINT SwapChainFlags) -> HRESULT
{
    return _this.ResizeBuffers(BufferCount, Width, Height, NewFormat, SwapChainFlags);
}